sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+  // WARNING: The type "Models.BlobStartCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobStartCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  startCopyFromURL(copySource: string, options?: BlobStartCopyFromURLOptions): Promise<Models.BlobStartCopyFromURLResponse>;
+  // WARNING: The type "Models.BlobCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  syncCopyFromURL(copySource: string, options?: BlobSyncCopyFromURLOptions): Promise<Models.BlobCopyFromURLResponse>;
+  // WARNING: The type "Models.BlobUndeleteResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobUndeleteResponse" needs to be exported by the package (e.g. added to index.ts)
+  undelete(options?: BlobUndeleteOptions): Promise<Models.BlobUndeleteResponse>;
+  withSnapshot(snapshot: string): BlobClient;
+}
+
+// @public
+interface BlobCreateSnapshotOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+  metadata?: Metadata;
+}
+
+// @public
+interface BlobDeleteOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+  deleteSnapshots?: Models.DeleteSnapshotsOptionType;
+}
+
+// @public
+interface BlobDownloadOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+  maxRetryRequests?: number;
+  progress?: (progress: TransferProgressEvent) => void;
+  rangeGetContentCrc64?: boolean;
+  rangeGetContentMD5?: boolean;
+  snapshot?: string;
+}
+
+// @public
+interface BlobGetPropertiesOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+}
+
+// @public
+interface BlobReleaseLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobRenewLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+class BlobSASPermissions {
+  add: boolean;
+  create: boolean;
+  delete: boolean;
+  static parse(permissions: string): BlobSASPermissions;
+  read: boolean;
+  toString(): string;
+  write: boolean;
+}
+
+// @public
+interface BlobSASSignatureValues {
+  blobName?: string;
+  cacheControl?: string;
+  containerName: string;
+  contentDisposition?: string;
+  contentEncoding?: string;
+  contentLanguage?: string;
+  contentType?: string;
+  expiryTime?: Date;
+  identifier?: string;
+  ipRange?: IPRange;
+  permissions?: string;
```

Comment:
Should this take something shaped like Blob/AccountSASPermissions so I can pass an object like `{ read: true, write: false }` instead of 'rw'?

Reply:
Concern: set of permissions changes over time. But it should be fine.